src/main/java/by/custom/utilcalculator/service/UserProgressManager.java

```diff
@@ -1,72 +1,98 @@
+        fileUserProgressStorage = FileUserProgressStorage.getInstance();
     }
 
     public static UserProgressManager getInstance() {
-        return BotFieldsManagerHolder.BOT_FIELDS_MANAGER;
+        return BotFieldsManagerHolder.USER_PROGRESS_MANAGER;
     }
 
-    public String processCarOrigin(String command, UserProgress userProgress) {
-        switch (command) {
-            case Command.EAES -> userProgress.setCountryOrigin(CountryOrigin.EAES);
-            case Command.OTHER_COUNTRIES -> userProgress.setCountryOrigin(CountryOrigin.OTHER);
-        }
-        return messagesCreator.getCountryOrigin(userProgress);
+    public void createNewUserProgress(final String chatID) throws WritingUserProgressIntoFileException {
+        fileUserProgressStorage.create(chatID);
     }
 
-    public String processOwnerType(String command, UserProgress userProgress) {
-        switch (command) {
-            case Command.JURIDICAL_PERSON -> userProgress.setOwnersType(OwnersType.JURIDICAL);
-            case Command.PHYSICAL_PERSON -> userProgress.setOwnersType(OwnersType.PHYSICAL);
-        }
-        return messagesCreator.getCountryOrigin(userProgress);
+    public String processCarOrigin(final String command, final String chatID) throws UserFileNotFoundException, ReadingUserProgressFromFileException, WritingUserProgressIntoFileException {
+        UserProgress userProgress;
+            userProgress = fileUserProgressStorage.get(chatID);
+            switch (command) {
+                case Command.EAES -> userProgress.setCountryOrigin(CountryOrigin.EAES);
+                case Command.OTHER_COUNTRIES -> userProgress.setCountryOrigin(CountryOrigin.OTHER);
+            }
+            fileUserProgressStorage.save(userProgress);
+            return messagesCreator.getCountryOrigin(userProgress);
     }
 
-    public String processCarAge(String command, UserProgress userProgress) {
-        switch (command) {
-            case Command.LESS_3_YEARS_AGE -> userProgress.setCarAge(CarAge.LESS_3_YEARS);
-            case Command.BETWEEN_3_AND_7_YEARS_AGE -> userProgress.setCarAge(CarAge.BETWEEN_3_AND_7_YEARS);
-            case Command.MORE_7_YEARS_AGE -> userProgress.setCarAge(CarAge.MORE_7_YEARS);
-        }
-        return messagesCreator.getCountryOrigin(userProgress);
+    public String processOwnerType(final String command, final String chatID) throws UserFileNotFoundException, ReadingUserProgressFromFileException, WritingUserProgressIntoFileException {
+        UserProgress userProgress;
+            userProgress = fileUserProgressStorage.get(chatID);
+            switch (command) {
+                case Command.JURIDICAL_PERSON -> userProgress.setOwnersType(OwnersType.JURIDICAL);
+                case Command.PHYSICAL_PERSON -> userProgress.setOwnersType(OwnersType.PHYSICAL);
+            }
+            fileUserProgressStorage.save(userProgress);
+            return messagesCreator.getCountryOrigin(userProgress);
```

Comment:
Are you sure that `getCountryOrigin` should be there?

Reply:
From this method starts to build string with a user choise. I have added a wrapper-method to imrove this code. It works fine, but naming was really confusing